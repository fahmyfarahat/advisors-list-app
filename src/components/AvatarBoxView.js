import React from 'react';
import PropTypes from 'prop-types';
import {
  Grid,
  Badge, 
  Box,
  Avatar,
  Typography
} from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';

const OnlineBadgeStyles = withStyles(theme => ({
  badge: {
    backgroundColor: '#44b700',
    boxShadow: `0 0 0 2px ${theme.palette.background.paper}`,
    '&::after': {
      position: 'absolute',
      top: 0,
      left: 0,
      width: '100%',
      height: '100%',
      borderRadius: '50%',
      animation: '$ripple 1.5s infinite ease-in-out',
      border: '1px solid #44b700',
      content: '""',
    },
  },
  '@keyframes ripple': {
    '0%': {
      transform: 'scale(.8)',
      opacity: 1,
    },
    '100%': {
      transform: 'scale(2.4)',
      opacity: 0,
    },
  },
}))(Badge);

const AvatarBoxView = ({ src, name, status }) => (
  <Grid container spacing={1}>
    <Grid item>
      <Box>
        <OnlineBadgeStyles
          invisible={status === 'offline' ? true : false}
          overlap="circle"
          anchorOrigin={{
            vertical: 'bottom',
            horizontal: 'right',
          }}
          variant="dot"
        >
          <Avatar alt={name} src={src} />
        </OnlineBadgeStyles>
      </Box>
    </Grid>
    <Grid item>
      <Typography variant="subtitle1">{name}</Typography>
    </Grid>
  </Grid>
);

AvatarBoxView.propTypes = {
  status: PropTypes.oneOf(['online', 'offline']).isRequired,
  src: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

export default AvatarBoxView;