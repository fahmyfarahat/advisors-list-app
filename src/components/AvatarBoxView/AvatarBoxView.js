import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '@material-ui/core/styles/withStyles';
import {
  Grid,
  Badge, 
  Box,
  Avatar,
  Typography
} from '@material-ui/core';

const OnlineBadgeStyles = withStyles(theme => ({
  badge: {
    backgroundColor: '#44b700',//TODO:: use theme colors
    boxShadow: `0 0 0 2px ${theme.palette.background.paper}`,
    '&::after': {
      position: 'absolute',
      top: 0,
      left: 0,
      width: '100%',
      height: '100%',
      borderRadius: '50%',
      animation: '$ripple 1.5s infinite ease-in-out',
      border: '1px solid #44b700',//TODO:: use theme colors
      content: '""',
    },
  },
  '@keyframes ripple': {
    '0%': {
      transform: 'scale(.8)',
      opacity: 1,
    },
    '100%': {
      transform: 'scale(2.4)',
      opacity: 0,
    },
  },
}))(Badge);

const AvatarBoxView = ({ src, name, status }) => (
  <Grid container spacing={1}>
    <Grid item>
      <Box>
        <OnlineBadgeStyles
          invisible={status === 'offline' ? true : false}
          overlap="circle"
          anchorOrigin={{
            vertical: 'bottom',
            horizontal: 'right',
          }}
          variant="dot"
        >
          <Avatar alt={name} src={src} />
        </OnlineBadgeStyles>
      </Box>
    </Grid>
    <Grid item>
      <Typography variant="subtitle1">{name}</Typography>
    </Grid>
  </Grid>
);

AvatarBoxView.propTypes = {
  status: PropTypes.oneOf(['online', 'offline']).isRequired,
  src: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

export default AvatarBoxView;