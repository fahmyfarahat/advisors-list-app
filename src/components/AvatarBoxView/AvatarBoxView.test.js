import React from "react";
import { shallow } from "enzyme";
import AvatarBoxView from "./AvatarBoxView";
import toJson from "enzyme-to-json";

describe("AvatarBoxView - general ", () => {
  it("should match Snapshot", () => {
    const props = {
      status: "online", 
      src: "url", 
      name: "name", 
    };
    const wrapper = shallow(<AvatarBoxView {...props} />);
    expect(wrapper).toHaveLength(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
