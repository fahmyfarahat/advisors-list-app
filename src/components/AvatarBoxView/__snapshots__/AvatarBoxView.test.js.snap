// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarBoxView - general  should match Snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={1}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(MuiBox)>
      <WithStyles(WithStyles(ForwardRef(Badge)))
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "bottom",
          }
        }
        invisible={false}
        overlap="circle"
        variant="dot"
      >
        <WithStyles(ForwardRef(Avatar))
          alt="name"
          src="url"
        />
      </WithStyles(WithStyles(ForwardRef(Badge)))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <WithStyles(ForwardRef(Typography))
      variant="subtitle1"
    >
      name
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
